ledger/ledger-on-sql/src/main/scala/com/daml/ledger/on/sql/SqlLedgerReaderWriter.scala

```diff
@@ -214,24 +203,32 @@ class SqlLedgerReaderWriter(
 }
 
 object SqlLedgerReaderWriter {
-  val FirstIndex: Index = 1
+  val StartIndex: Index = 1
 
-  private val FirstOffset: Offset = Offset(Array(FirstIndex))
+  private val StartOffset: Offset = Offset(Array(StartIndex))
 
-  def apply(
+  def owner(
       ledgerId: LedgerId,
       participantId: ParticipantId,
       jdbcUrl: String,
   )(
       implicit executionContext: ExecutionContext,
       materializer: Materializer,
       logCtx: LoggingContext,
-  ): Future[SqlLedgerReaderWriter] =
-    Future {
-      val database = Database(jdbcUrl)
-      new SqlLedgerReaderWriter(ledgerId, participantId, database)
-    }.map { ledger =>
-      ledger.migrate()
-      ledger
+  ): ResourceOwner[SqlLedgerReaderWriter] =
+    for {
+      dispatcher <- ResourceOwner.forCloseable(
+        () =>
+          Dispatcher(
+            "sql-participant-state",
+            zeroIndex = StartIndex,
+            headAtInitialization = StartIndex,
```

Comment:
Always using `StartIndex` for `headAtInitialization` will break when starting the ledger with a non-empty non-in-memory database, but it's probably good enough for now.

Reply:
Oh, good point, this'll be me not understanding the API correctly. Will fix in a different PR.

Reply:
I think it'd be good to have a conformance test that somehow restarts the ledger and makes sure everything is A-OK.

Reply:
That's not a conformance/ledger api test though, because we cannot remotely restart the ledger. But +1 on having such a test specifically for the KV SQL Ledger.

Reply:
The conformance test Bazel task knows how to start and stop the ledger, and therefore how to restart it. :-)